cardano-api/src/Cardano/Api/KeysByron.hs

```diff
@@ -54,6 +68,127 @@ import           Cardano.Api.SerialiseTextEnvelope
 -- This is a type level tag, used with other interfaces like 'Key'.
 --
 data ByronKey
+data ByronKeyLegacy
+
+class IsByronLegacyFormat key where
```

Comment:
Naming: later on you use this class to generalise from ByronKey to any Byron key. But the name `IsByronLegacyFormat key =>` suggests that key is a byron legacy format, but that's not what you mean: you mean it's a Byron key type, and the class can tell you if the format is legacy or modern.

So I'd suggest instead:
```
class IsByronKey key where
    byronKeyFormat :: ByronKeyFormat key
```